clippy_lints/src/misc.rs

```diff
@@ -224,7 +244,23 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for Pass {
                                                initref=initref));
                 }
             );
-        }}
+        }};
+        if_let_chain! {[
+            let StmtSemi(ref expr, _) = s.node,
+            let Expr_::ExprBinary(ref binop, ref a, ref b) = expr.node,
+            binop.node == BiAnd || binop.node == BiOr,
+            let Some(sugg) = Sugg::hir_opt(cx, a),
```

Comment:
I have no idea. The provided helper assume that this might fail, and offer two versions, one returning an `Option` and one using a default value (usually ".." from what I've read). While it would be appropriate to show `if f() { ..; }` as a suggestion, I would not like `if .. { g(); }`, hence the use of the `_opt` variant for `a` and the default value for `b`.